### ISSUE TITLE
An issue about the String.prototype.replace().

### ISSUE THREAD

**Issue by NWU-NISL** [mozilla/rhino#738]
**Revision**

rhino-1.7.12

**Test case**

```javascript
var NISLFuzzingFunc = function(e,nislMutationParameter0){
	return e.replace(/>/,nislMutationParameter0);
};
var NISLParameter0 = '>';
var nislMutationArgument0 = '$+';
var NISLCallingResult = NISLFuzzingFunc(NISLParameter0,nislMutationArgument0);
print(NISLCallingResult);
```

**Execution steps**

```javascript
java -jar rhino/rhino-1.7.12.jar -debug -version 200 testcase.js
```

**Output**

```javascript
The output is empty.
```

**Expected behavior**

```javascript
$+
```

**Description**

According to ES standard,  '$+' should replace the original string as it is, but rhino does not.

**Comment by tonygermano** [comment]
That's because Rhino implements [RegExp.lastParen](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastParen).

[String.proto.replace()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter) states to use `$$` to insert a literal "$".

If you use `var nislMutationArgument0 = '$$+';` it will work as expected.

**Comment by p-bakker** [comment]
@gbrail you tagged this as `good first issue`, but what's the task at hand here? disabling the RegExp.lastParen feature when using Context.VERSION_ES6?

**Comment by tonygermano** [comment]
This proposal has been at stage 3 for a few years: https://github.com/tc39/proposal-regexp-legacy-features

**Comment by p-bakker** [comment]
That proposal amends Annex B of the EcmaScript spec, which are additions that are considered NOT tree be part of the spec, but stuff that web browsers out to implement (for compatibility reasons)

As such, Rhino, not being a web browser, shouldn't implement them.

I think however that Rhino might have implemented many of Annex B, with It's origin as a Java port of SpiderMonkey

**Comment by tonygermano** [comment]
Yeah, I had shared that more as a guide to assist in figuring out a way forward considering those legacy properties are already implemented.
